pkg/apis/application/application.go

```diff
@@ -113,7 +113,7 @@ func (app Application) OnSync() error {
 	return nil
 }
 
-func (app *Application) updateRadixRegistrationStatus(rr *v1.RadixRegistration, changeStatusFunc func(currStatus *v1.RadixRegistrationStatus)) error {
+func (app Application) updateRadixRegistrationStatus(rr *v1.RadixRegistration, changeStatusFunc func(currStatus *v1.RadixRegistrationStatus)) error {
```

Comment:
```suggestion
func (app *Application) updateRadixRegistrationStatus(rr *v1.RadixRegistration, changeStatusFunc func(currStatus *v1.RadixRegistrationStatus)) error {
```
We should use pointer receiver on all methods for Application